### PR TITLE
Integrate starknet devnet

### LINKED ISSUE #23: integrate starknet-devnet
https://github.com/Shard-Labs/starknet-devnet/tree/master

### PR DESCRIPTION
**Pull request by martriay**
Fixes #23. Depends on #26.

- [x] install it on `nile install`
- [x] add `deploy` method
- [x] handle deployments (simple address:artifact mapping)
- [x] add `invoke` method
- [x] add `call` method
- [x] add `node` command
- [x] update readme
- [x] prepare [blogpost](https://medium.com/@martriay/manage-your-starknet-deployments-with-nile-%EF%B8%8F-e849d40546dd)

### CODE REVIEW

File: src/nile/commands/init.py [hunk 1]
```diff
@@ -25,95 +26,9 @@ def init_command():
 
     # create project directories
     print("🗄  Creating project directory tree")
-    create_contracts()
-    create_tests()
-
-    with open("Makefile", "w") as fp:
-        fp.write(makefile)
+    copy_tree(os.path.join(os.path.dirname(__file__), "../base_project"), ".")
```
Comment (fracek) [inline src/nile/commands/init.py:29]: Maybe using `pathlib.Path` would make this cleaner? E.g. `Path(__file__).parent / 'base_project')`

Comment (martriay) [reply src/nile/commands/init.py:29]: thanks! will fix this in #26

File: src/nile/commands/node.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+"""Command to start StarkNet local network."""
+import subprocess
+
+
+def node_command():
+    """Start StarkNet local network."""
+    subprocess.check_call("starknet-devnet")
```
Comment (fracek) [inline src/nile/commands/node.py:7]: Should it catch `FileNotFoundError` and recommend to install `starknet-devnet` using `nile install`?

**Review by fracek — COMMENTED** [review]
I believe calling the current test network `mainnet` will be confusing once main net will be deployed at the end of the month. I suggest changing the usage of `mainnet` to `testnet`.

**Comment by martriay** [comment]
> I believe calling the current test network `mainnet` will be confusing once main net will be deployed at the end of the month. I suggest changing the usage of `mainnet` to `testnet`.

Do you think so? I thought the goerli deployment behaved as mainnet as of today, such that if in the future you use the `starknet` command with these params, it will send txs to mainnet. I was under the impression goerli is just the settlement layer but the starknet nodes were already "the main network".

**Comment by fracek** [comment]
That's true, we can update the names as they come. Then it all looks good to me!

**Pull request merged**